test/testpushnotifications.cpp

```diff
@@ -0,0 +1,187 @@
+#include <QTest>
+#include <QVector>
+
+#include "account.h"
+#include "common.h"
+#include "pushnotifications.h"
+#include "creds/abstractcredentials.h"
+
+using namespace std::chrono_literals;
+
+class WebSocketStub : public OCC::AbstractWebSocket
+{
+    Q_OBJECT
+public:
+    void open(const QUrl & /*url*/) override
+    {
+        emit connected();
+    }
+    void close() override { }
+    qint64 sendTextMessage(const QString &message) override
+    {
+        ++sendTextMessageCalledCount;
+        sendTextMessageCalledArguments.append(message);
+        return 0;
+    }
+
+    uint32_t sendTextMessageCalledCount = 0;
```

Comment:
Isn't it exactly `sendTextMessageCalledArguments.size()` ?

Reply:
You're right:)